server/src/main/scala/actors/queues/FlightsRouterActor.scala

```diff
@@ -153,65 +128,43 @@ class FlightsRouterActor(
-    val eventualUpdatedMinutesDiff: Source[Seq[MillisSinceEpoch], NotUsed] =
-      Source(groupByTerminalAndDay(container))
-        .mapAsync(1) {
-          case ((terminal, day), updates) =>
-            handleUpdateAndGetDiff(terminal, day, updates)
+  override def partitionUpdates: PartialFunction[FlightUpdates, Map[(Terminal, UtcDate), FlightUpdates]] = {
+    case container: FlightsWithSplits =>
+      Map()
+    case container: SplitsForArrivals =>
+      container.splits
+        .groupBy {
+          case (uniqueArrival, _) => (uniqueArrival.terminal, SDate(uniqueArrival.scheduled).toUtcDate)
+        }
+        .map {
+          case (terminalDay, allSplits) => (terminalDay, SplitsForArrivals(allSplits))
         }
-    combineEventualDiffsStream(eventualUpdatedMinutesDiff)
-  }
 
-  def groupByTerminalAndDay(container: FlightsWithSplitsDiff): Map[(Terminal, UtcDate), FlightsWithSplitsDiff] = {
-    val updates: Map[(Terminal, UtcDate), List[ApiFlightWithSplits]] = container.flightsToUpdate
-      .groupBy(flightWithSplits => (flightWithSplits.apiFlight.Terminal, SDate(flightWithSplits.apiFlight.Scheduled).toUtcDate))
-    val removals: Map[(Terminal, UtcDate), List[UniqueArrival]] = container.arrivalsToRemove
-      .groupBy(ua => (ua.terminal, SDate(ua.scheduled).toUtcDate))
-
-    val keys = updates.keys ++ removals.keys
-    keys
-      .map { terminalDay =>
-        val diff = FlightsWithSplitsDiff(updates.getOrElse(terminalDay, List()), removals.getOrElse(terminalDay, List()))
-        (terminalDay, diff)
-      }
-      .toMap
+    case container: ArrivalsDiff =>
+      val updates: Map[(Terminal, UtcDate), Iterable[Arrival]] = container.toUpdate.values
+        .groupBy(arrivals => (arrivals.Terminal, SDate(arrivals.Scheduled).toUtcDate))
+      val removals: Map[(Terminal, UtcDate), Iterable[Arrival]] = container.toRemove
+        .groupBy(arrival => (arrival.Terminal, SDate(arrival.Scheduled).toUtcDate))
+
+      val keys = updates.keys ++ removals.keys
+      keys
+        .map { terminalDay =>
+          val terminalUpdates = updates.getOrElse(terminalDay, List())
+          val terminalRemovals = removals.getOrElse(terminalDay, List())
+          val diff = ArrivalsDiff(terminalUpdates, terminalRemovals)
+          (terminalDay, diff)
+        }
+        .toMap
   }
 
-  private def combineEventualDiffsStream(eventualUpdatedMinutesDiff: Source[Seq[MillisSinceEpoch], NotUsed]): Future[Seq[MillisSinceEpoch]] = {
-    eventualUpdatedMinutesDiff
-      .fold(Seq[MillisSinceEpoch]())(_ ++ _)
-      .runWith(Sink.seq)
-      .map {
-        case containers if containers.nonEmpty => containers.reduce(_ ++ _)
-        case _ => Seq[MillisSinceEpoch]()
-      }
-      .recover {
-        case t =>
-          log.error(t, "Failed to combine containers")
-          Seq[MillisSinceEpoch]()
-      }
-  }
+  def affectsFromUpdate(partition: (Terminal, UtcDate), updates: FlightUpdates): Future[UpdatedMillis] =
```

Comment:
as an example, this could be `daysAffectedByUpdate`

Reply:
Ok, I've gone with `millisAffectedByUpdate`